#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_use_ifexists_in_default_pa.dpatch by CJ van den Berg <cj@vdbonline.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: use .ifexists meta command to load optional modules only if available

@DPATCH@

diff -Nru pulseaudio-0.9.5-orig/src/daemon/default.pa.in pulseaudio-0.9.5/src/daemon/default.pa.in
--- pulseaudio-0.9.5-orig/src/daemon/default.pa.in	2006-08-25 14:08:11.000000000 +0200
+++ pulseaudio-0.9.5/src/daemon/default.pa.in	2006-10-06 02:32:56.761435138 +0200
@@ -26,12 +26,18 @@
 #load-module module-null-sink
 #load-module module-pipe-sink
 
+.ifexists /usr/lib/pulse-0.9/modules/module-hal-detect.so
+
 ### Automatically load driver modules depending on the hardware available
 load-module module-hal-detect
 
+.else
+
 ### Alternatively use the static hardware detection module (for systems that
 ### lack HAL support
-#load-module module-detect
+load-module module-detect
+
+.endif
 
 ### Load audio drivers automatically on access
 #add-autoload-sink output module-oss device="/dev/dsp" sink_name=output source_name=input
@@ -41,8 +47,14 @@
 #add-autoload-sink output module-alsa-sink sink_name=output
 #add-autoload-source input module-alsa-source source_name=input
 
-### Load several protocols
+.ifexists /usr/lib/pulse-0.9/modules/module-esound-protocol-unix.so
+
+### Load esound protocol
 load-module module-esound-protocol-unix
+
+.endif
+
+### Load native protocol
 load-module module-native-protocol-unix
 
 ### Network access (may be configured with paprefs, so leave this commented
